src/api/spec/models/token/workflow_spec.rb

```diff
@@ -159,5 +159,47 @@
 
       it { expect { subject }.to change(workflow_token, :triggered_at) & change(workflow_run, :response_url).to('https://api.github.com') }
     end
+
+    context 'with a ping event' do
+      let(:scm) { 'github' }
+      let(:event) { 'ping' }
+      let(:github_payload) { { sender: { url: 'https://api.github.com' } } }
+      let(:github_extractor_payload) do
+        {
+          scm: 'github',
+          event: 'ping',
+          api_endpoint: 'https://api.github.com'
+        }
+      end
+      let(:scm_extractor) { TriggerControllerService::SCMExtractor.new(scm, event, github_payload) }
+      let(:scm_webhook) { SCMWebhook.new(payload: github_extractor_payload) }
+      let(:workflow) do
+        Workflow.new(scm_webhook: scm_webhook, token: workflow_token,
+                     workflow_instructions: { steps: [branch_package: { source_project: 'home:Admin', source_package: 'ctris', target_project: 'dev:tools' }] })
+      end
+      let(:workflows) { [workflow] }
+
+      before do
+        # Skipping call since it's tested in the Workflow model
+        allow(workflow).to receive(:call).and_return(true)
+
+        allow(TriggerControllerService::SCMExtractor).to receive(:new).with(scm, event, github_payload).and_return(scm_extractor)
+        allow(scm_extractor).to receive(:call).and_return(scm_webhook)
+        allow(SCMStatusReporter).to receive(:new).and_return(proc { true })
+      end
+
+      subject { workflow_token.call(workflow_run: workflow_run, scm_webhook: scm_extractor.call) }
+
+      it 'returns before checking for validation errors' do
+        expect(subject).to be_empty
+      end
+
+      it { expect { subject }.to change(workflow_token, :triggered_at) & change(workflow_run, :response_url).to('https://api.github.com') }
```

Comment:
```suggestion
      it { expect { subject }.to change(workflow_token, :triggered_at).and change(workflow_run, :response_url).to('https://api.github.com') }
```

There are some other parts of our code where we use the `&` operator. But we should make use of `and` to chain expectations. See https://relishapp.com/rspec/rspec-expectations/v/3-12/docs/compound-expectations#use-%60and%60-to-chain-expectations

This suggestion, still not tested, should also work.